src/MahApps.Metro/MahApps.Metro.Shared/Controls/Dialogs/DialogManager.cs

```diff
@@ -375,7 +406,8 @@ public static Task HideMetroDialogAsync(this MetroWindow window, BaseMetroDialog
                 {
                     window.RemoveDialog(dialog);
 
-                    return HandleOverlayOnHide(settings,window);
+                    settings = settings ?? dialog.DialogSettings;
```

Comment:
Is this ok, or should it be `settings = settings ?? (dialog.DialogSettings ?? window.MetroDialogOptions);`